lib/rubocop/cop/rspec/subject_declaration.rb

```diff
@@ -24,24 +24,22 @@ class SubjectDeclaration < Base
         MSG_LET = 'Use subject explicitly rather than using let'
         MSG_REDUNDANT = 'Ambiguous declaration of subject'
 
-        # @!method let_subject?(node)
-        def_node_matcher :let_subject?, <<~PATTERN
-          (send nil? {#Helpers.all} {(sym :subject) (str "subject")} ...)
-        PATTERN
-
-        # @!method redundant_subject?(node)
-        def_node_matcher :redundant_subject?, <<~PATTERN
-          (send nil? {#Subjects.all} {(sym :subject) (str "subject")} ...)
+        # @!method offensive_subject_declaration?(node)
+        def_node_matcher :offensive_subject_declaration?, <<~PATTERN
+          (send nil? ${#Subjects.all #Helpers.all} {(sym :subject) (str "subject")} ...)
         PATTERN
 
         def on_send(node)
-          let_subject?(node) do
-            add_offense(node, message: MSG_LET)
-          end
+          offense = offensive_subject_declaration?(node)
+          return unless offense
+
+          add_offense(node, message: message_for(offense))
```

Comment:
```suggestion
          offensive_subject_declaration?(node) do |offense|
            add_offense(node, message: message_for(offense))
          end
```
Also the `?` in the name could be removed, as we are using it in a non-boolean context